### PR TITLE
Handle recoverable exceptions in message processing workers.

### PR DESCRIPTION
**Pull request by wwajerowicz**

### CODE REVIEW

File: atlas-api/src/main/java/org/atlasapi/messaging/KafkaMessagingModule.java [hunk 1]
```diff
@@ -37,7 +41,7 @@ public MessageSenderFactory messageSenderFactory() {
     @Override
     @Bean
     public MessageConsumerFactory<KafkaConsumer> messageConsumerFactory() {
-        return new KafkaMessageConsumerFactory(zookeeper, messagingSystem);
+        return new KafkaMessageConsumerFactory(zookeeper, messagingSystem, backOffIntervalMillis, maxBackOffMillis);
```
Comment (tfm) [inline atlas-api/src/main/java/org/atlasapi/messaging/KafkaMessagingModule.java]: This should be a `Duration` not a `Long`

**Comment by mbst-jamie** [comment]
All fine

File: atlas-processing/src/main/java/org/atlasapi/messaging/EquivalentScheduleStoreScheduleUpdateWorker.java [hunk 2]
```diff
@@ -40,7 +41,7 @@ public void process(ScheduleUpdateMessage message) {
                 timer.stop();
             }
         } catch (WriteException e) {
-            log.error("update failed for " + message.getScheduleUpdate(), e);
+            throw new RecoverableException("update failed for " + message.getScheduleUpdate(), e);
```
Comment (tfm) [inline atlas-processing/src/main/java/org/atlasapi/messaging/EquivalentScheduleStoreScheduleUpdateWorker.java]: Not sure it's up to the exception type to decide if it's recoverable or not. Shouldn't the exception type reflect the type of error, not whether it's recoverable or not?

**Comment by tfm** [comment]
back to you, @wwajerowicz

File: atlas-processing/src/main/java/org/atlasapi/messaging/EquivalentScheduleStoreScheduleUpdateWorker.java (hunk 2 above)
Comment (wwajerowicz) [inline atlas-processing/src/main/java/org/atlasapi/messaging/EquivalentScheduleStoreScheduleUpdateWorker.java]: @tfm but doesn't the type of error actually reflect if it's recoverable or not?

**Pull request merged**